app/src/main/java/com/andrew/tasky/presentation/task_detail/TaskDetailFragment.kt

```diff
@@ -4,29 +4,198 @@ import androidx.lifecycle.ViewModelProvider
     private lateinit var navController: NavController
     private lateinit var binding: FragmentTaskDetailBinding
-
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-        viewModel = ViewModelProvider(this).get(TaskDetailViewModel::class.java)
-    }
+    private lateinit var taskType: TaskType
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding = FragmentTaskDetailBinding.bind(view)
         navController = Navigation.findNavController(view)
+        viewModel = ViewModelProvider(this).get(TaskDetailViewModel::class.java)
+        taskType = TaskType.values()[arguments?.getInt("taskType")!!]
+
+        subscribeToObservables()
+        setupTaskType()
+
+        if(viewModel.getEditMode()){
+            editMode()
+        }
+        else{
+            nonEditMode()
+        }
+
+        binding.closeButton.setOnClickListener {
+            navController.popBackStack()
+        }
+
+        binding.editButton.setOnClickListener {
+            viewModel.setEditMode(true)
+            editMode()
+        }
+
+        binding.saveButton.setOnClickListener {
+            navController.popBackStack()
+        }
+
+        binding.taskTitle.setOnClickListener {
+            val editType = EditType.TITLE.ordinal
+            val bundle = bundleOf("editType" to editType)
+            navController.navigate(R.id.action_taskDetailFragment_to_editFragment, bundle)
+        }
+
+        binding.editTitleButton.setOnClickListener {
+            val editType = EditType.TITLE.ordinal
+            val bundle = bundleOf("editType" to editType)
+            navController.navigate(R.id.action_taskDetailFragment_to_editFragment, bundle)
+        }
+
+        binding.editDescriptionButton.setOnClickListener {
+            val editType= EditType.DESCRIPTION.ordinal
+            val bundle = bundleOf("editType" to editType)
+            navController.navigate(R.id.action_taskDetailFragment_to_editFragment, bundle)
+        }
+
+        binding.descriptionTextView.setOnClickListener {
+            val editType= EditType.DESCRIPTION.ordinal
+            val bundle = bundleOf("editType" to editType)
+            navController.navigate(R.id.action_taskDetailFragment_to_editFragment, bundle)
+        }
+
+        binding.time.setOnClickListener {
+            timePickerFragment()
+        }
+
+        binding.date.setOnClickListener {
+            datePickerFragment()
+        }
+    }
+
+    private fun subscribeToObservables() {
+        lifecycleScope.launchWhenStarted {
+            viewModel.selectedDate.collectLatest {
+                binding.date.text = it
+            }
+        }
+        lifecycleScope.launchWhenStarted {
+            viewModel.selectedTime.collectLatest {
+                binding.time.text = it
+            }
+        }
+    }
+
+    private fun timePickerFragment(){
```

Comment:
Naming recommendation: `showTimePickerFragment()` since that's what it does right? 😄 
Same for datePickerFragment() below